### DISCUSSION TITLE
Problem with Password Reset API call using Fortify, Sanctum and Laravel 11

### DISCUSSION THREAD

**Discussion by davidwbsc** [laravel/framework#51118] [Q&A] [answered]
I'm setting up a new Laravel 11 project for a Vue frontend via an API.

I have got the login and log out working correctly, but when I try the reset password endpoint (/forgot-password), I receive a 500 error and `Route [password.reset] not defined.`

From the Fortify docs it seems like this should work out of the box. Is there anything I am missing?

Thanks

**Comment by joy2362** [comment]
Add the following code in your fortify service provider boot method and  check if its working or not also make sure blade file exits


```
Fortify::resetPasswordView(function (Request $request) {
        return view('auth.reset-password', ['request' => $request]);
    });
```

**Reply by davidwbsc** [reply]
Thank you for your reply.

I'm not using blade templates as I'm using an API. So I am calling the /forgot-password endpoint (Sorry that wasn't clear in my initial question). 

(Just added the correct endpoint to my initial question)

**Reply by joy2362** [reply, marked as answer]
This issue occurs because you may have set views as false and it removes password reset route and that's why it's not able to generate password reset url to solve this issue you can add these line in you app service provider boot method

`use Illuminate\Auth\Notifications\ResetPassword;`

and 

`
  ResetPassword::createUrlUsing(function ($user, string $token) {
                return 'http://127.0.0.1:8000/admin/reset-password/'.$token;
            });
`


![image](https://github.com/laravel/framework/assets/49846472/f743a351-1f9a-4ba5-98c8-65dd1ab852fd)

**Reply by davidwbsc** [reply]
Thank you, that was it! It makes so much sense now, as Laravel has no idea what the front end route of my password reset would be.

In case anyone else is interest, this is the exact code I used:

```
ResetPassword::createUrlUsing(
    function ($notifiable, $token) {
        return config('app.frontend_url') . '/reset-password/' . $token . '/' . $notifiable->getEmailForPasswordReset();
    });
```

**Reply by joy2362** [reply]
Good to know it solve your issue

**Reply by yousifDR4** [reply]
thanks for the solv and question
